example-app/App.tsx

```diff
@@ -39,15 +39,54 @@ const Header = () => {
   );
 };
 
+type DeepLinkResult =
+  | {
+      success: true;
+    }
+  | {
+      success: false;
+      errorReason: string;
+    }
+  | undefined;
+
 const App = () => {
+  const [result, setResult] = React.useState<DeepLinkResult>(undefined);
+
+  React.useEffect(() => {
+    Linking.addEventListener('url', event => {
+      setResult(handleDeepLink(event.url));
+    });
+
+    async function getDeepLink() {
+      let url = await Linking.getInitialURL();
+      setResult(handleDeepLink(url));
+    }
+
+    getDeepLink();
+  }, []);
+
+  const handleDeepLink = (url: string | null): DeepLinkResult => {
+    if (!url) {
+      return;
+    }
+
+    let deeplinkUrl = new URL(url, true);
+    let error = deeplinkUrl.query.error;
+    if (error) {
+      return {success: false, errorReason: error!};
```

Comment:
```suggestion
      return {success: false, errorReason: error};
```

not needed? You check it in the if statement on the line above.

Reply:
Sloppy, sorry